src/wagtail_ai/static_src/api.tsx

```diff
@@ -0,0 +1,38 @@
+import type { ApiUrlName, WagtailAiConfiguration } from './custom';
+
+class APIRequestError extends Error {}
+
+export const getAIConfiguration = (): WagtailAiConfiguration => {
+  const configurationElement =
+    document.querySelector<HTMLScriptElement>('#wagtail-ai-config');
+  if (!configurationElement || !configurationElement.textContent) {
+    throw new Error('No wagtail-ai configuration found.');
+  }
+
+  try {
+    return JSON.parse(configurationElement.textContent);
+  } catch (err) {
+    throw new SyntaxError(
+      `Error parsing wagtail-ai configuration: ${err.message}`,
+    );
+  }
+};
+
+export const fetchResponse = async (
+  action: keyof typeof ApiUrlName,
+  body: FormData,
```

Comment:
**Question:** Out of curiosity, why do we prefer FormData instead of JSON? Is that so we could send files in the future or is that what is a custom in Wagtail?

Reply:
I was wondering this too, but went with what the text completion endpoint was already doing.

Reply:
Feels like an extra step to parse JSON in the view, whereas with FormData we're just passing `request.POST` to the form?